### PR TITLE
chore: split fetching all TEs from fetching paginated TEs [TECH-1645]

### PR DESCRIPTION
**Pull request by enricocolasante**
Continuing the work already done for `enrollments` (https://github.com/dhis2/dhis2-core/pull/15434) to `tracked entities`
* Split fetching all tracked entities from all tracked entity pages by extracting a `PageParam` class that contains the defaults for `page`, `pageSize` and uses primitive types `int`, `boolean`.
* The controller will then either fetch all or a page of tracked entities based on `skipPaging`

# Next PR(s)

Push the pagination logic we currently have in the service into the repository. This will remove a source of error and make the code cleaner. As right now the repository fetches `pageSize+1` if `totalPages=false` to see if the page is the last page. The service then removes that extra page. All of that will live in the repository.

### CODE REVIEW

File: dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/export/OrderAndPaginationExporterTest.java [hunk 1]
```diff
@@ -145,41 +144,28 @@ void shouldReturnPaginatedTrackedEntitiesGivenNonDefaultPageSize()
...
-        // when
-        // handleLastPageFlag was copied from the event service which works in conjunction with the
-        // event store
-        // that fetches pageSize + 1 when totalCount=false. This split of logic between
-        // service/store is
-        // error prone and hard to follow. We will fix/refactor it for all entities so this is
-        // purely a concern
-        // of the store.
-        () -> assertSlimPager(1, 3, true, firstPage.getPager()),
+        () -> assertPager(2, 3, firstPage),
```
Comment (enricocolasante) [inline dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/export/OrderAndPaginationExporterTest.java:154]:
```suggestion
        () -> assertPager(1, 3, firstPage),
```

**Review by muilpp — APPROVED** [review]

**Review by lucaCambi77 — APPROVED** [review]
